framework/src/fvkernels/FVElementalKernel.C

```diff
@@ -0,0 +1,109 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVElementalKernel.h"
+#include "MooseVariableFV.h"
+#include "Assembly.h"
+#include "SubProblem.h"
+#include "NonlinearSystemBase.h"
+
+#include "libmesh/elem.h"
+
+#include "metaphysicl/raw_type.h"
+
+InputParameters
+FVElementalKernel::validParams()
+{
+  InputParameters params = FVKernel::validParams();
+  params.registerSystemAttributeName("FVElementalKernel");
+  params += MaterialPropertyInterface::validParams();
+  return params;
+}
+
+FVElementalKernel::FVElementalKernel(const InputParameters & parameters)
+  : FVKernel(parameters),
+    MooseVariableInterface(this,
+                           false,
+                           "variable",
+                           Moose::VarKindType::VAR_NONLINEAR,
+                           Moose::VarFieldType::VAR_FIELD_STANDARD),
+    CoupleableMooseVariableDependencyIntermediateInterface(this, false),
+    MaterialPropertyInterface(this, blockIDs(), Moose::EMPTY_BOUNDARY_IDS),
+    _var(*mooseVariableFV()),
+    _u(_var.adSln()),
+    _current_elem(_assembly.elem())
+{
+  addMooseVariableDependency(&_var);
+}
+
+// Note the lack of quadrature point loops in the residual/jacobian compute
+// functions. This is because finite volumes currently only works with
+// constant monomial elements. We only have one quadrature point regardless of
+// problem dimension and just multiply by the element volume.
+
+void
+FVElementalKernel::computeResidual()
+{
+  prepareVectorTag(_assembly, _var.number());
+  _local_re(0) += MetaPhysicL::raw_value(computeQpResidual() * _assembly.elem()->volume());
```

Comment:
Important to note that `_assembly.elem()->volume()` builds a quadrature rule for every evaluation

Reply:
I'm mistaken for simpler elements. But: it does do some evaluation.

Do you have the volumetric reinit done during this loop?

Reply:
Yeah - I'll fix this.